sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -524,10 +524,12 @@ class annotated_ref {
   public:
     annotated_ref(const annotated_ref&) = delete;
     operator T() const;
-    T operator=(T) const;
+    template <typename TT>
+    T operator=(TT&&) const;
```

Comment:
Why are we adding perfect forwarding?  Can you give an example where this is better than `operator=(T)`?

Reply:
Hi Greg. We were considering if T is a non-trivially copyable type. If we do pass-by-value, we might have side effects there as it might invoke a bunch of copy constructor. We think to do a perfect forwarding might be the best option.

Reply:
I think `T` does need to be trivially copyable for the reasons I mentioned in the email thread.  Most of the properties only make sense for a trivially copyable type.  For example, the cache hints assume that reads and writes of `T` are simple memory load / store operations.  It makes no sense to apply these cache hints to a function call.

Do we have some real-life use case where `T` is not trivially copyable and has some associated property?

If we do allow `T` to be non-trivially copyable, I think we will need to start documenting each property to indicate whether it can be used with non-trivially-copyable types.

Reply:
I am not sure if we have a real use case for non-trivially copyable type right now. If user want to stream a buffer array to kernel using pointers I guess they might go with pipes but not kernel args.

But I think we should not stop users from doing that in the future. The current annotated_ptr header is NOT ready for taking non-trivially-copyable types and we might want to support that in the future. This perfect forwarding impl is fixing the volatile type issue, but and the meantime address the LValue and RValue issue in advance.

For the non-trivially copyable type. I think we should document it in the spec, right? Roland suggested putting a warning when T is non-trivially copyable. But I am wondering if we clearly say non-trivially copyable is not supported in the spec, should we doing error or warning?

Reply:
Just send out another email. I think if we both agree on not supporting non-trivially copyable type then we can just do pass-by-value here

Reply:
Updated wth Spec with "Explicitly state that use non-trivially-copyable type in annotated_ptr is not recommended"

and do pass-by-value